selectel/resource_selectel_domains_domain_v1.go

```diff
@@ -0,0 +1,111 @@
+package selectel
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/selectel/domains-go/pkg/v1/domain"
+)
+
+func resourceDomainsDomainV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDomainsDomainV1Create,
+		Read:   resourceDomainsDomainV1Read,
+		Delete: resourceDomainsDomainV1Delete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"create_date": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"change_date": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"user_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDomainsDomainV1Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+	client := config.domainsV1Client()
+
+	createOpts := &domain.CreateOpts{
+		Name: d.Get("name").(string),
+	}
+
+	log.Print(msgCreate(objectDomain, createOpts))
+	domainObj, _, err := domain.Create(ctx, client, createOpts)
+	if err != nil {
+		return errCreatingObject(objectDomain, err)
+	}
+
+	d.SetId(strconv.Itoa(domainObj.ID))
+
+	return resourceDomainsDomainV1Read(d, meta)
+}
+
+func resourceDomainsDomainV1Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+	client := config.domainsV1Client()
+
+	log.Print(msgGet(objectDomain, d.Id()))
+
+	domainID, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("failed to parse domain ID: %w", err)
```

Comment:
Maybe we can move this repeated message into a separate wrapper. Example: https://github.com/terraform-providers/terraform-provider-selectel/blob/master/selectel/errors.go#L29